src/lib/operators/abstract_join_operator.cpp

```diff
@@ -57,18 +57,20 @@ std::string AbstractJoinOperator::description(DescriptionMode description_mode)
     return "Column #"s + std::to_string(column_id);
   };
 
-  const auto* const separator = description_mode == DescriptionMode::MultiLine ? "\n" : " ";
-
+  const char separator = (description_mode == DescriptionMode::SingleLine ? ' ' : '\n');
   std::stringstream stream;
-  stream << AbstractOperator::description(description_mode) << separator << "(" << _mode << " Join where "
-         << column_name(true, _primary_predicate.column_ids.first) << " " << _primary_predicate.predicate_condition
-         << " " << column_name(false, _primary_predicate.column_ids.second);
+  stream << AbstractOperator::description(description_mode) << separator;
+  stream << "(" << _mode << " Join where ";
+  stream << column_name(true, _primary_predicate.column_ids.first) << " ";
+  stream << _primary_predicate.predicate_condition << " ";
+  stream << column_name(false, _primary_predicate.column_ids.second);
 
-  // add information about secondary join predicates
+  // Add information about secondary join predicates
   for (const auto& secondary_predicate : _secondary_predicates) {
-    stream << " AND " << column_name(true, secondary_predicate.column_ids.first) << " "
-           << secondary_predicate.predicate_condition << " "
-           << column_name(false, secondary_predicate.column_ids.second);
+    stream << separator << "AND ";
```

Comment:
Is that change for MultiLineMode intended? From always ` AND `, we now have `\nAND ` for MultiLineMode which is fine for me.

Reply:
Could you post an example output plan?

Reply:
> Is that change for MultiLineMode intended? 

Yes. For example:

<img width="530" alt="image" src="https://user-images.githubusercontent.com/9380917/131553973-9c2d3f83-11c6-4ea2-8af4-25e9c7e39b09.png">

Reply:
We could capitalize `where` as well?

Reply:
Maybe even better:

```
JoinHash
(<join_mode>)
WHERE <predicate>
AND <predicate>
```

Reply:
To consider: 
<img width="735" alt="image" src="https://user-images.githubusercontent.com/9380917/131557152-33ea1340-eaa0-4286-97e0-242e2c6f2b11.png">

Reply:
Personally, I think I would consider

```
JoinHash (<join_mode>)
WHERE <predicate1>
AND <predicate2>
```

to be the best option.

Reply:
<img width="970" alt="image" src="https://user-images.githubusercontent.com/9380917/131644119-e8e81730-bb00-4fde-a5f9-9df863ceb628.png">

Reply:
Looks good to me, @Bouncner any opinion on that?

Reply:
I like the formatting, but I don't like the WHERE here. WHERE is SQL and filters tuples out, that do not match.
A join predicate can mean something very different (e.g., for outer joins which tuples to keep even though there is no match or anti-joins where is means the opposite).

Reply:
Then `ON` or omit at all?

Reply:
Good question. I would say omit as `ON` is also pretty SQL'y and there is no way one can use `ON` to generate semi joins.

I tend to say omit, but I don't have a strong opinion here.

Reply:
Same for me, no strong opinion but I would omit it.

Reply:
Alright, I removed WHERE.